### DISCUSSION TITLE
🔊 Tones

### DISCUSSION THREAD

**Discussion by matiasandina** [KravitzLabDevices/FED3#10] [General]
I was taking a closer look at the multiple ways that we have for delivering sound from the FEDs.
I see most programs in the `Examples` default to `ConditionedStimulus()` with the default length of `duration = 200` ms. This also has the `pixelsOn()` call which may or may not be desired.
But I also see `GoNoGo.ino` making nice use of `Click()` and `Tone()`.

I guess I just wanted to kick off discussion to see when people are using what, if they are changing duration/frequency. Lights vs no lights. Curious to see if people found that some combinations just don't work for conditioning (or are too slow).

Also, any thoughts about loudness of the tones? 

Code might come handy here:

```
void FED3::ConditionedStimulus(int duration) {
  tone (BUZZER, 4000, duration);
  pixelsOn(0,0,10,0);  //blue light for all
}

void FED3::Click() {
  tone (BUZZER, 800, 8);
}

void FED3::Tone(int freq, int duration){
  tone (BUZZER, freq, duration);
}

void FED3::Noise(int duration) {
  // White noise to signal errors
  for (int i = 0; i < duration/50; i++) {
    tone (BUZZER, random(50, 250), 50);
    delay(duration/50);
  }
}
```

**Comment by KravitzLab** [comment]
Hey Matias!  Thanks for starting this!  But want to post it on the Google group?  I think I'm the only one who looks at this haha :)

**Comment by Sophi** [comment]
I look at it too :)
